core/space/services/services_sharing.go

```diff
@@ -86,18 +87,25 @@ func (s *Space) GenerateFileSharingLink(
 	)
 }
 
+// uploads the shared file to ipfs through users public bucket in hub
 func (s *Space) uploadSharedFileToIpfs(
 	ctx context.Context,
 	password string,
 	sharedContent io.Reader,
 	fileName string,
 	bucketName string,
 ) (domain.FileSharingInfo, error) {
-	fileUploadResult := s.ic.AddItem(ctx, sharedContent)
-	if err := fileUploadResult.Error; err != nil {
-		return EmptyFileSharingInfo, errors.Wrap(err, "encrypted file upload failed")
+	b, err := s.tc.GetPublicShareBucket(ctx)
+	if err != nil {
+		return EmptyFileSharingInfo, errors.Wrap(err, "failed to get public files bucket")
 	}
-	encryptedFileHash := fileUploadResult.Resolved.Cid().String()
+
+	uploadResult, _, err := b.UploadFile(ctx, fmt.Sprintf("%s-%d", fileName, time.Now().UnixNano()), sharedContent)
```

Comment:
I'm currently getting an rpc error at this line:
```
 rpc error: code = Unknown desc = instance not found
```
I think it has to do with something not yet created on the hubs thread, other this, this is ready for review.